### ISSUE TITLE
Changing Device Registration

### ISSUE THREAD

**Issue by orionkoepke** [NilsG-S/antifreeze-back-end#36]
We might consider changing the way the devices registers with the server. Instead of having the the device take in the user's user name and password and pass those to the server, the device could generate a UUID (Universal Unique Identifier) and then give it to the user and back-end on registration.

Steps:
1. The device hosts a web server for the user to put in the wifi information.
2. The user puts in the wifi information and hits a "submit and download" button.
3. Upon hitting submit the web server generates a UUID.
4. The server then sends the generated UUID to the user as a download.
5. The server registers to the back-end with the UUID.
6. The user logs in to the front-end to register the device.
7. The user supplies the UUID they downloaded earlier as the device they would like to register. 

Advantages:
1. It keeps the device server from needing to deal with things such as incorrect user name and password.

Disadvantages:
1. The UUID has an extremely low chance, but still a chance, that it will be the exact same as another generated UUID.
- Since the chances of this happening are extremely low, this edge case could potential be ignored. If it were to happen the user would just set up the device again, so the device would generate a new UUID.

Other Things To Consider:
1. While the device would not have to deal with things such as user name and password errors for the back end, it would still have to deal with wifi errors such as incorrect wifi password.

**Comment by NilsG-S** [comment]
Depending on how this is implemented, it could introduce a fair amount of complexity. On the back-end side of things, I would have to alter the device registration system and account for "unverified" devices. The front-end would have to add device registration as well. However, I don't think my work would take more than 3-4 hours.

My main question is whether this would provide sufficient benefit to balance the extra work. As you said, the problem doesn't entirely go away. All this would do is reduce the number of failure points (email, password, WiFi password vs WiFi password). I could definitely see the utility for a real version of this app, but not as much for the scope of this class.

Would this change provide large benefits to the device team?

**Comment by orionkoepke** [comment]
I do not think that this will provide enough benefit at this point to the system to justify the extra work. However, like you said, it would definitely be worth implementing for the real version.
